FHIR 101 - Practical Guide.ipynb

```diff
@@ -1497,11 +1499,246 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "fetch('http://hapi.fhir.org/baseR4/StructureDefinition/591026/$snapshot')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "##### Examining a specific instance of a resource\n",
+    "The general format for looking at a specific instance is:\n",
+    "\n",
+    "`GET [base]/[type]/[id]`\n",
+    "\n",
+    "This will return the full payload for a resource with a specific ID."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "fetch('http://hapi.fhir.org/baseR4/Observation/1286681')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "##### References\n",
+    "📘 For more information on references, refer to the [FHIR docs section](https://www.hl7.org/fhir/search.html#reference)\n",
+    "\n",
+    "FHIR resources can reference each other. For example, an `Observation` may reference a specific `Patient`. The FHIR search API allows us to search for resources that contain a specific reference. As an example, let's fetch all `Observations` that refer to a `Patient` with an id of 1286630:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "fetch('http://hapi.fhir.org/baseR4/Observation?subject=Patient/1286630')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "##### Filtering resources\n",
+    "Earlier, we saw that the general format for a query with search parameters is:\n",
+    "\n",
+    "`GET [base]/[type]?name=value&...`\n",
+    "\n",
+    "Here are some examples of search queries utilizing research use cases:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "# Example - fetch all Patients that are alive\n",
+    "fetch('http://hapi.fhir.org/baseR4/Patient?deceased=false')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "# Fetch all Conditions that were observed in female Patients\n",
+    "# This is an example of chaining\n",
+    "fetch('http://hapi.fhir.org/baseR4/Condition?patient.gender=female')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "# Fetch all Patients that have multiple sclerosis\n",
+    "# This is an example of reverse chaining\n",
+    "fetch('http://hapi.fhir.org/baseR4/Patient?_has:Condition:patient:code=G35')"
```

Comment:
Can we refer to a doc explaining about why `Condition:code` is an invalid syntax? (`Condition:code` may seem more logically intuitive than `Condition:patient:code`to people who are new to FHIR, but familiar with REST.

Reply:
Good idea